app/server/route-generator.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+/**
+ * Generates express-jefferson route configs from a methods.js and middlewares.js
+ * config file pair, suitable for placement in the 'routes' config block.
+ *
+ * This function will marry an independent name hash with a middleware hash to create a jefferson-compatible 'routes' config block.
+ * The return value is therefore a map of the form:
+ * {
+ *   <path>: {
+ *      <method>: [<middlewares>]
+ *   }
+ * }
+ *
+ * The intention of this is to allow separation of method naming and HTTP config from middleware definitions for use elsewhere (such as link generation).
+ *
+ * The names should be a map of the form:
+ * {
+ *   <name>: {
```

Comment:
Yes, it is the latest. I'm actually reverting it intentionally here, so that I *can* have named methods for use with the link builder (rather than having to reference them by path, which is the key, and thus replicating the path template everywhere).